Update docker setup steps - Closes #1063 (#1064)
LiskHQ/lisk-docs: modules/ROOT/pages/setup/docker.adoc

:url_index_usage: index.adoc#usage
:url_setup: setup/index.adoc
:url_core_setup_npm: v3@lisk-core::setup/npm.adoc
:url_config: configuration/docker.adoc
:url_management: management/docker.adoc
:sectnums:

How to set up Lisk Service and connect it to a local Lisk Core node.

== Setting up Lisk Core

> Lisk Service is installed in addition to a Lisk blockchain application like Lisk Core, in order to provide enriched network data about the network for third party services.

Reviewer: ```suggestion
Lisk Service is installed in addition to a Lisk blockchain application such as Lisk Core, in order to provide enriched network data for third-party services.
```

The first step is therefore to install Lisk Core.

> We recommend using NPM to install Lisk Core conveniently through the command line:

Reviewer: ```suggestion
It is recommended to use NPM to conveniently install Lisk Core through the command line:
```

[source,bash]
----
npm install --global --production lisk-core
----

> Check out the xref:{url_core_setup_npm}[Lisk Core NPM setup] guide to find more information how to install Lisk Core on your machine.

Reviewer: ```suggestion
Check out the xref:{url_core_setup_npm}[Lisk Core NPM setup] guide for more information on how to install Lisk Core on your machine.
```

== Configuring Lisk Core

Use the Lisk Core CLI to start Lisk Core with the desired configurations.

The `--network` flag defines which blockchain network Lisk Core will connect to.

`--api-ws` enables the WebSocket API of the node.

> `--api-ws-port` allows to define the port for the WebSocket connection.

Reviewer: ```suggestion
`--api-ws-port` allows the user to define the port for the WebSocket connection.
```

====

==== Open necessary ports

> Make sure to have the port open which allows external services communication with the Lisk Service APIs.

Reviewer: ```suggestion
Ensure the port which allows external services communication with the Lisk Service APIs is open.
```

----
git clone https://github.com/LiskHQ/lisk-service.git
cd lisk-service
git checkout v0.3.0
----

== Build Lisk Service

Build Lisk Service from local files:

./lisk-service/
[source,bash]
----
make build
----

Lisk Service is now ready to use on your machine.

== Connecting Lisk Service to Lisk Core

> To connect Lisk Service to a local Lisk Core node, set the environment variable `LISK_CORE_WS` before you start Lisk Service like this:

Reviewer: ```suggestion
To connect Lisk Service to a local Lisk Core node, set the environment variable `LISK_CORE_WS` before starting Lisk Service as shown below:
```

.Inside of the `lisk-service` root folder
[source,bash]
----
export LISK_CORE_WS=ws://localhost:8080
make up
----

> Alternatively, it is possible to define a file with all desired environment variables and use it to start Lisk Service like this:

Reviewer: ```suggestion
Alternatively, it is possible to define a file with all the desired environment variables and use it to start Lisk Service as shown below:
```

./lisk-service/
[source,bash]
----
cd docker/network

> vim my-env.env

Reviewer 2: Shall we advice people to use `nano` which is default in most distros?

Update: just checked, these files are not that relevant anymore.
The default file path is `docker/network/core3-default.env` and it is not advised to change its content.

User can create a `my-env` file but it's important in more complicated deployments only.

`LISK_CORE_WS`, `ENABLE_HTTP_API`, `ENABLE_WS_API` are the only supported params by default.

However, if we want to mention it at all, the current text is good to go 👍

Author: Ok np, I now commented out the section about the env file, and updated the above section about exporting the env variables accordingly.

----

Define key-value pairs of all desired environment variables, for example:

TIP: Press kbd:[i] to enter the insert mode in vim.

./lisk-service/docker/network/my-env.env
[source,bash]
----
# Lisk ecosystem configuration
LISK_CORE_WS=ws://localhost:8080

# Enabled Gateway APIs
ENABLE_HTTP_API=http-status,http-version2
ENABLE_WS_API=blockchain,rpc-v2
----

TIP: Save and quit vim again by first leaving the insert mode with kbd:[Esc] and then by typing kbd:[:wq] and kbd:[Enter].

Now start Lisk Service with the following command:

./lisk-service/
[source,bash]
----
make up-custom-my-env
----

> This will start Lisk Service with all environment variables which you previousely defined in `my-env.env`.

Reviewer: ```suggestion
This will start Lisk Service with all the environment variables previously defined in `my-env.env`.
```

----
git clone https://github.com/LiskHQ/lisk-service.git
cd lisk-service
git checkout v0.3.0
----

== Build Lisk Service

Build Lisk Service from local files:

./lisk-service/
[source,bash]
----
make build
----

Lisk Service is now ready to use on your machine.

== Connecting Lisk Service to Lisk Core

To connect Lisk Service to a local Lisk Core node, set the environment variable `LISK_CORE_WS` before starting Lisk Service as shown below:

.Inside of the `lisk-service` root folder
[source,bash]
----
export LISK_CORE_WS=ws://localhost:8080
#export ENABLE_HTTP_API=http-status,http-version2
#export ENABLE_WS_API=blockchain,rpc-v2
# Command to start Lisk Service
make up
----

> <<<<<<< HEAD

Author: true, removed it now
